src/HDF5.jl

```diff
@@ -283,7 +283,7 @@ end
 convert(::Type{hid_t}, attr::Attribute) = attr.id
 show(io::IO, attr::Attribute) = isvalid(attr) ? print(io, "HDF5 attribute: ", name(attr)) : print(io, "HDF5 attribute (invalid)")
 
-mutable struct Attributes
```

Comment:
Is this the only `mutable struct` -> `struct` audited?

Reply:
I didn't check thoroughly — I noticed it while browsing through your rename PR and this one was the only one that stood out to me as not needing to be mutable since it doesn't add a finalizer.

Reply:
It looks like the only candidate, though:
```
$ git grep 'mutable struct'
src/HDF5.jl:mutable struct File <: DataFile
src/HDF5.jl:mutable struct Group <: DataFile
src/HDF5.jl:mutable struct Properties
src/HDF5.jl:mutable struct Dataset
src/HDF5.jl:mutable struct Datatype
src/HDF5.jl:mutable struct Dataspace
src/HDF5.jl:mutable struct Attribute
```